integration-tests/tests/src/iterators.ts

```diff
@@ -0,0 +1,198 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { expect } from "chai";
+
+import {
+    Dog,
+    DogSchema,
+    IDog,
+    IPerson,
+    Person,
+    PersonSchema
+} from "./schemas/person-and-dogs";
+
+describe("Iterating", () => {
+    let realm: Realm;
+    let dogs: { [name: string]: Dog };
+    let persons: { [name: string]: Person };
+
+    beforeEach(() => {
+        // Add linkingObjects to the PersonAndDogs schema
+        const PersonWithDogsSchema = {
+            ...PersonSchema,
+            properties: {
+                ...PersonSchema.properties,
+                dogs: {
+                    type: "linkingObjects",
+                    objectType: "Dog",
+                    property: "owner"
+                }
+            }
+        };
+        realm = new Realm({ schema: [DogSchema, PersonWithDogsSchema] });
+        realm.write(() => {
+            persons = {
+                alice: realm.create<IPerson>("Person", {
+                    name: "Alice",
+                    age: 16
+                }) as Person,
+                bob: realm.create<IPerson>("Person", {
+                    name: "Bob",
+                    age: 42
+                }) as Person,
+                charlie: realm.create<IPerson>("Person", {
+                    name: "Charlie",
+                    age: 62
+                }) as Person
+            };
+            dogs = {
+                max: realm.create<IDog>("Dog", {
+                    age: 1,
+                    name: "Max",
+                    owner: persons.alice
+                }) as Dog,
+                rex: realm.create<IDog>("Dog", {
+                    age: 3,
+                    name: "Rex",
+                    owner: persons.alice
+                }) as Dog,
+                bobby: realm.create<IDog>("Dog", {
+                    age: 5,
+                    name: "Bobby",
+                    owner: persons.bob
+                }) as Dog
+            };
+            // Make Bob and Charlie frieds
+            persons.bob.friends.push(persons.charlie);
+            persons.charlie.friends.push(persons.bob);
+        });
+    });
+
+    it("returns an instance of Results", () => {
+        const results = realm.objects("Person");
+        expect(results).instanceOf(Realm.Results);
+    });
+
+    it("throws if object schema doesn't exist", () => {
+        expect(() => {
+            realm.objects("SomeOtherClass");
+        }).throws("Object type 'SomeOtherClass' not found in schema.");
+    });
+
+    type CollectionCallback = () => Realm.Collection<any>;
+
+    function itCanIterate(
+        getCollection: CollectionCallback,
+        expectedNames: string[]
+    ) {
```

Comment:
function itCanIterate(
        getCollection: CollectionCallback,
        expectedNames: string[]) 
{

little and tiny code formatting note :)

Reply:
Not with the current `tslint` rules :)

Reply:
that sucks that our rules require us to do that.

Reply:
Personally I don't really care either way, as long as we have a standard that I can use automated tools to format my code into. Our [tslint.json](https://github.com/realm/realm-js/blob/master/integration-tests/tests/tslint.json) mostly use the recommended / latest options, in an effort to make us less special compared to other code-bases on the interwebs.